gcp_variant_transforms/options/variant_transform_options.py

```diff
@@ -89,10 +91,25 @@ def add_arguments(self, parser):
               'variants. This is useful when there are header fields in '
               'variants not defined in the header sections.'))
 
+  def validate(self, parsed_args):
+    try:
+      # Gets at most one pattern match result of type `filesystems.MatchResult`.
+      first_match = FileSystems.match([parsed_args.input_pattern], [1])[0]
+      if not first_match.metadata_list:
+        raise ValueError(
+            'File pattern {} returned '
+            'no matched files.'.format(parsed_args.input_pattern))
+    except BeamIOError as e:
```

Comment:
@arostamianfar : This only checks `storage.objects.list` permission as that is what is required for the match condition. To check `storage.objects.get` (which is what a file read specifically uses I think) we would probably need to do a read of the match result. Additionally, if files matching the input pattern have different permissions (some readable, some not readable) the validation could pass if the first file matched has acceptable permissions.

Reply:
I think those edge cases are ok and we're doing "best effort" here. In all problematic cases that I know of, the users had a typo in their GCS bucket path, which resulted in a cryptic http error message, so they thought something was wrong with our code.